app/uk/gov/hmrc/gform/formtemplate/FormTemplatesControllerRequestHandler.scala

```diff
@@ -157,7 +162,35 @@ object FormTemplatesControllerRequestHandler {
         case (None, Some(_), None, None)       => JsSuccess(())
       }
 
-    sectionValidations andKeep jsonValue.transform(
+    val dmsFormIdValidations: JsResult[Unit] =
+      (jsonValue \ "destinations").toOption match {
+        case Some(id) =>
+          val errorMessages = id
+            .as[JsArray]
+            .value
+            .map { jv =>
+              ((jv \ "id").as[String], (jv \ "type").as[String], (jv \ "dmsFormId").toOption.map(_.as[String]))
+            }
+            .map {
+              // format: off
+              case (_, "hmrcDms", Some(dmsFormId))
+                  if dmsFormId.toString.length > 0 && dmsFormId.toString.length <= 12 => ""
```

Comment:
Probably a good idea to trim the value before checking as a space seems to get through fine `"dmsFormId": " "`

Reply:
Good catch. Addressed.